src/ext/java/org/opentripplanner/ext/siri/AddedTripBuilder.java

```diff
@@ -0,0 +1,379 @@
+
+    // Create the "scheduled version" of the trip
+    var aimedStopTimes = new ArrayList<StopTime>();
+    for (int stopSequence = 0; stopSequence < calls.size(); stopSequence++) {
+      StopTime stopTime = createStopTime(
+        trip,
+        departureDate,
+        stopSequence,
+        calls.get(stopSequence),
+        stopSequence == 0,
+        stopSequence == (calls.size() - 1)
+      );
+
+      aimedStopTimes.add(stopTime);
+    }
+
+    // TODO: We always create a new TripPattern to be able to modify its scheduled timetable
+    StopPattern stopPattern = new StopPattern(aimedStopTimes);
+    TripPattern pattern = TripPattern
+      .of(getTripPatternId.apply(trip))
+      .withRoute(trip.getRoute())
+      .withMode(trip.getMode())
+      .withNetexSubmode(trip.getNetexSubMode())
+      .withStopPattern(stopPattern)
+      .build();
+
+    TripTimes tripTimes = new TripTimes(trip, aimedStopTimes, transitModel.getDeduplicator());
+    tripTimes.setServiceCode(transitModel.getServiceCodes().get(trip.getServiceId()));
+    pattern.add(tripTimes);
+    TripTimes updatedTripTimes = new TripTimes(tripTimes);
+
+    // Loop through calls again and apply updates
+    for (int stopSequence = 0; stopSequence < calls.size(); stopSequence++) {
+      TimetableHelper.applyUpdates(
+        departureDate,
+        updatedTripTimes,
+        stopSequence,
+        stopSequence == (calls.size() - 1),
+        isJourneyPredictionInaccurate,
+        calls.get(stopSequence),
+        occupancy
+      );
+    }
+
+    if (cancellation || stopPattern.isAllStopsCancelled()) {
+      updatedTripTimes.cancelTrip();
+    } else {
+      updatedTripTimes.setRealTimeState(RealTimeState.ADDED);
+    }
+
+    /* Validate */
+    var validityResult = updatedTripTimes.validateNonIncreasingTimes();
+    if (validityResult.isFailure()) {
+      return validityResult.toFailureResult();
+    }
+
+    // Adding trip to index necessary to include values in graphql-queries
+    // TODO - SIRI: should more data be added to index?
+    transitModel.getTransitModelIndex().getTripForId().put(tripId, trip);
+    transitModel.getTransitModelIndex().getPatternForTrip().put(trip, pattern);
+    transitModel.getTransitModelIndex().getPatternsForRoute().put(route, pattern);
+
+    return Result.success(new TripUpdate(stopPattern, updatedTripTimes, serviceDate));
+  }
+
+  /**
+   * Method to create a Route. Commonly used to create a route if a realtime message
+   * refers to a route that is not in the transit model.
+   *
+   * We will find the first Route with same Operator, and use the same Authority
+   * If no operator found, copy the agency from replaced route
+   *
+   * If no name is given for the route, an empty string will be set as the name.
+   *
+   * @return a new Route
+   */
+  private Route createRoute() {
+    var routeBuilder = Route.of(entityResolver.resolveId(lineRef));
+
+    routeBuilder.withShortName(shortName);
```

Comment:
The long and short names should not be equal, or even contain parts of each other (at least per GTFS spec). Is the `PublishedLineName` which of these? ShortName is exposed as `publicCode` in the API.

Reply:
In the SIRI-profile, the field `PublishedLineName` is defined to be: "Public number or name of the line." - so it may be both `publicCode` and `name`.

Also - it looks like the result I was looking for is already handled in `route.getName()`:  https://github.com/opentripplanner/OpenTripPlanner/blob/dev-2.x/src/main/java/org/opentripplanner/transit/model/network/Route.java#L197-L199